go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -512,18 +512,36 @@ func (be *BuiltinBackupEngine) backupFiles(
 		wg.Add(1)
 		go func(i int) {
 			defer wg.Done()
-
-			// Wait until we are ready to go, skip if we already
-			// encountered an error.
-			sema.Acquire(ctx, 1)
+			fe := &fes[i]
+			// Wait until we are ready to go, return if we encounter an error
+			acqErr := sema.Acquire(ctx, 1)
+			if acqErr != nil {
+				log.Errorf("not able to acquire semaphore to backup file:%s  err: %s", fe.Name, acqErr.Error())
+				bh.RecordError(acqErr)
+				return
+			}
 			defer sema.Release(1)
+			// check for context cancellation explicitly because, the way semaphore code is
```

Comment:
Nit, but we should use sentence capitalization and punctuation.

Reply:
done